qp/pdf.py

```diff
@@ -55,18 +60,54 @@ def quantize(self, percent=10.0, number=None):
         print("Result: ", self.quantiles)
         return self.quantiles
 
-    def interpolate(self):
+    def interpolate(self, number=100):
+        """
+        Interpolates the quantiles.
 
-        return
+        Parameters
+        ----------
+        number: int
+            The number of points over which to interpolate, bounded by the quantile value endpoints
+
+        Returns
+        -------
+        grid: ndarray, float
+            The input grid upon which to interpolate
+        interpolated : ndarray, float
+            The interpolated points.
+
+        Notes
+        -----
 
-    def plot(self, limits):
+        """
+        if self.interpolator is None:
+
+            if self.quantiles is None:
+                self.quantiles = self.quantize()
+            self.difs = self.quantiles[1:]-self.quantiles[:-1]
+            self.mids = (self.quantiles[1:]+self.quantiles[:-1])/2.
```

Comment:
I implemented extrapolation, sort of.  It's just linear extrapolation forbidding negative values, since this is to be applied to probability distributions.

I also separated out interpolate(), which returns a function, from approximate(x), which returns the evaluation of the interpolator at point(s) x.  Is that any better?